doc/FAQ.md

```diff
@@ -0,0 +1,104 @@
+in the SparkConf or spark-defaults.conf.
+
+### Where should I set configuration options for the connector?
+
+The suggested location is to use the `spark-defaults.conf` file in your spark/conf directory but 
+this file is ONLY used by spark-submit. Any applications not running through spark submit will ignore
+this file. 
+ 
+For those you should set the options in the `SparkConf` object used to 
+create your `SparkContext`. Usually this will take the form of a series of statements that look like
+
+    val conf = SparkConf()
+      .set("Option","Value")
+      ...
+      
+    val sc = SparkContext(conf)
+
+### Why are my write tasks timing out/ failing?
+
+The most common cause of this is that Spark is able to issue write requests much more quickly than
+Cassandra can handle them. This can lead to GC issues and build up of hints. If this is the case
+with your application try lowering the number of concurrent writes and the current batch size using
+the following options.
+
+   spark.cassandra.output.batch.size.rows
+   spark.cassandra.output.concurrent.writes
+   
+### Why are my executors throwing `OutOfMemoryException`s while Reading from Cassandra?
+
+This is usually means that the size of the partitions you are attempting to create are larger than
+the executor's heap can handle. Remember that all of the executors run in the same JVM so the size
+of the data is multiplied by the number of executor slots.
+
+To fix this either increase the heap size of the executors `spark.executor.memory`
+ or shrink the size of the partitions by decreasing `spark.cassandra.input.split.size`
+ 
+### Why can't my spark job find My Application Classes / Anonymous Functions?
+ 
+This occurs when your application code hasn't been placed on the classpath of the Spark Executor. When using
+Spark Submit make sure that the jar contains all of the classes and dependencies for running your code. 
+To build a fat jar look into using sbt assembly, or look for instructions for your build tool of choice.
+
+If you are not using the recommended approach with Spark Submit, make sure that your dependencies 
+have been set in the `SparkConf` using `setJars` or by distributing the jars yourself and modifying 
+the executor classpath.
+ 
+### Why don't my case classes work? 
+Usually this is because they have been defined within another object/class. Try moving the definition
+outside of the scope of other classes.
+ 
+### Why can't my spark job connect to Cassandra?
+
+Check that your Cassandra instance is on and responds to cqlsh. Make sure that the rpc address also
+accepts incoming connections on the interface you are setting as `rpc_address` in the cassandra.yaml file.
+Make sure that you are setting the `spark.cassandra.connection.host` property to the interface which
+the rpc_address is set to.
+
+When troubleshooting it is sometimes useful to set this to `0.0.0.0` so any incoming connection will
```

Comment:
to set this -> to set rpc_address (be more explicit here, as it is unnclear whether you're referring to spark.cassandra.connection.host property or rpc_address in the previous sentence).